pkg/registry/generic/etcd/etcd.go

```diff
@@ -441,6 +450,9 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	// delete immediately, or no graceful deletion supported
 	out := e.NewFunc()
 	if err := e.Storage.Delete(ctx, key, out); err != nil {
+		if ignoreNotFound && storage.IsNotFound(err) {
+			return e.finalizeDelete(obj, true)
```

Comment:
> hmm, it won't be exactly the same if obj wasn't already deleted

Right.. I can modify the GuaranteedUpdate to return the updated object, but that sounds like another can of worm.